### DISCUSSION TITLE
Zoom/map dragging not clipping to map boundaries

### DISCUSSION THREAD

**Discussion by programtiago** [visgl/deck.gl#5874] [Q&A]
I used code from [this](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/mapbox) example.

https://user-images.githubusercontent.com/62779154/122063675-607f6300-cde8-11eb-8096-d2320e092455.mp4

**Comment by padawannn** [comment]
I don't really understand what the question is. In the video, I can see that base map tiles take some time to render but it could be because you're recording a video or your network is a bit slow. Could you send a Codepen example and confirm if this is your problem?

**Reply by NickAcPT** [reply]
By reading the title and watching the video, I think the problem is the white bar/background shown on the video @ 0:08.
The slow tile rendering is most likely due to the recording software.

Here's a screenshot from the video:
![Screenshot from the video](https://i.imgur.com/mGct2Ks.png)

**Reply by padawannn** [reply]
Thank you @NickAcPT
It seems as if some HTML item was pushing the map container down or changing its height, in fact, the eye icon disappears

**Reply by NickAcPT** [reply]
> Thank you @NickAcPT
> It seems as if some HTML item was pushing the map container down or changing its height, in fact, the eye icon disappears

@padawannn What I think is happening is that @programtiago is navigating the map with their mouse. As such, the vanishing of the eye icon is explained as it is the same colour as the background.

The map above has a small pitch value which is probably why the background shows in the first place, but what OP seems to want is to have the map to not show the background when moved around.

**Comment by programtiago** [comment]
It takes some time to render the layer because im dealing with large data tho. But my problem isn't that. It's about the white background div appears. But in [this](https://docs.mapbox.com/mapbox-gl-js/example/set-perspective/) example, and it seems that happens the same.

Thanks for your attention @padawannn
